crates/ide-completion/src/render.rs

```diff
@@ -1947,8 +1946,8 @@ fn main() {
 }
             "#,
             expect![[r#"
-                ex core::ops::Deref::deref(&T(S)) (use core::ops::Deref) [type_could_unify]
                 ex core::ops::Deref::deref(&t) (use core::ops::Deref) [type_could_unify]
+                ex core::ops::Deref::deref(&T(S)) (use core::ops::Deref) [type_could_unify]
                 lc m [local]
```

Comment:
Are these results non-deterministic-ish?

Reply:
Not really, will debug later why exactly the ordering changed.

Reply:
Would be great if you could take a quick look at that, will merge this given the ordering change being possibly non-deterministic shouldn't be related to this PR.

Reply:
Checked that collapsing terms before Cartesian product  https://github.com/rust-lang/rust-analyzer/pull/17203/files#diff-5797341fbf7c8fcf87bd0bfd50c533bff963b96d1185fadbda31cfebc114fdbaR130-R135 causes the order to change. Since I think there is no preferred order I think the current order is find.
If you got any good suggestions for preferred order then I'm happy to change it - would like to switch from `Vec` to `FxHashSet` anyway for its uniqueness guarantees, but couldn't really do it as it would cause "random" order